src/api/services/iuf/sessions_test.go

```diff
@@ -946,3 +947,205 @@ func TestProcessOutput(t *testing.T) {
+										Value: v1alpha1.AnyStringPtr("this_is_the_value_of_another_output"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantActivity: iuf.Activity{
+				Name: activity.Name,
+				OperationOutputs: map[string]interface{}{
+					"stage_params": map[string]interface{}{
+						"process-media": map[string]interface{}{
+							"products": map[string]interface{}{},
+						},
+					},
+				},
+				Products: activity.Products,
+			},
+			wantErr: true,
+		},
+		{
+			name: "two outputs - valid yaml but invalid iuf-manifest",
+			workflow: &v1alpha1.Workflow{
+				Status: v1alpha1.WorkflowStatus{
+					Nodes: v1alpha1.Nodes{
+						"this-is-a-name-of-templateRef": v1alpha1.NodeStatus{
+							DisplayName: "product-name-dash-operation-name",
+							Outputs: &v1alpha1.Outputs{
+								Parameters: []v1alpha1.Parameter{
+									{
+										Name:  "this_is_the_name_of_an_output",
+										Value: v1alpha1.AnyStringPtr("valid: yaml"),
+									},
+									{
+										Name:  "this_is_the_name_of_another_output",
+										Value: v1alpha1.AnyStringPtr("this_is_the_value_of_another_output"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantActivity: iuf.Activity{
+				Name: activity.Name,
+				OperationOutputs: map[string]interface{}{
+					"stage_params": map[string]interface{}{
+						"process-media": map[string]interface{}{
+							"products": map[string]interface{}{},
+						},
+					},
+				},
+				Products: activity.Products,
+			},
+			wantErr: false,
+		},
+		{
+			name: "two outputs - valid yaml but 'alomot' valid iuf-manifest",
+			workflow: &v1alpha1.Workflow{
+				Status: v1alpha1.WorkflowStatus{
+					Nodes: v1alpha1.Nodes{
+						"this-is-a-name-of-templateRef": v1alpha1.NodeStatus{
+							DisplayName: "product-name-dash-operation-name",
+							Outputs: &v1alpha1.Outputs{
+								Parameters: []v1alpha1.Parameter{
+									{
+										Name:  "this_is_the_name_of_product",
+										Value: v1alpha1.AnyStringPtr("name: this-is-a-name\nversion: this-is-a-version"),
+									},
+									{
+										Name:  "this_is_the_parent-directory",
+										Value: v1alpha1.AnyStringPtr("this_is_the_value_of_parent-directory"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantActivity: iuf.Activity{
+				Name: activity.Name,
+				OperationOutputs: map[string]interface{}{
+					"stage_params": map[string]interface{}{
+						"process-media": map[string]interface{}{
+							"products": map[string]interface{}{
+								"this-is-a-name": map[string]interface{}{
+									"parent_directory": "this_is_the_value_of_parent-directory",
+								},
+							},
+						},
+					},
+				},
+				Products: []iuf.Product{
+					{
+						Name:             "this-is-a-name",
+						Version:          "this-is-a-version",
+						OriginalLocation: "this_is_the_value_of_parent-directory",
+						Manifest:         `{"name":"this-is-a-name","version":"this-is-a-version"}`,
```

Comment:
where is the "Validated" field?

Reply:
nice catch ... so this go-cmp i used to compare objects... 
obj = nil
compare(obj,false) is going to be "true"

basically nil == false in this lib 

that said i should add validated here so others don't ask the same